extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -0,0 +1,152 @@
+import { IGenerateScriptSettings, IPublishSettings } from '../models/IPublishSettings';
+import { Project } from '../models/project';
+import { promptForPublishProfile } from './publishDatabaseDialog';
+
+/**
+ * Create flow for Publishing a database using only VS Code-native APIs such as QuickPick
+ */
+export async function launchPublishDatabaseQuickpick(project: Project): Promise<void> {
+
+	// 1. Select publish settings file (optional)
+	// TODO: Hook up to dacfx service
+	const browseProfileOption = await vscode.window.showQuickPick(
+		[constants.dontUseProfile, constants.browseForProfile],
+		{ title: constants.selectProfile, ignoreFocusOut: true });
+	if (!browseProfileOption) {
+		return;
+	}
+
+	// let publishSettingsFile: vscode.Uri | undefined;
+	if (browseProfileOption === constants.browseForProfile) {
+		const locations = await promptForPublishProfile(project.projectFolderPath);
+		if (!locations) {
+			return;
+		}
+		// publishSettingsFile = locations[0];
+	}
+
+	// 2. Select connection
+	// TODO: Hook up to MSSQL
+	const connectionProfile = await vscode.window.showQuickPick(
+		['Connection 1', 'Connection 2', 'Create New Connection'],
+		{ title: constants.selectConnection, ignoreFocusOut: true });
+	if (!connectionProfile) {
+		return;
+	}
+	const dbs = ['db1', 'db2'];
+	const dbQuickpicks = dbs.map(db => {
+		return {
+			label: db,
+			dbName: db
+		} as vscode.QuickPickItem & { dbName: string, isCreateNew?: boolean };
+	});
+	// Ensure the project name is an option, either adding it if it doesn't already exist or moving it to the top if it does
+	const projectNameIndex = dbs.findIndex(db => db === project.projectFileName);
+	if (projectNameIndex === -1) {
+		dbQuickpicks.unshift({ label: constants.newDatabaseTitle(project.projectFileName), dbName: project.projectFileName });
+	} else {
+		dbQuickpicks.splice(projectNameIndex, 1);
+		dbQuickpicks.unshift({ label: project.projectFileName, dbName: project.projectFileName });
+	}
+
+	dbQuickpicks.push({ label: constants.createNew, dbName: '', isCreateNew: true });
+	// 3. Select database
+	// TODO: Hook up to MSSQL
+	let databaseName = '';
+	while (databaseName === '') {
+		const selectedDatabase = await vscode.window.showQuickPick(
+			dbQuickpicks,
+			{ title: constants.selectDatabase, ignoreFocusOut: true });
+		if (!selectedDatabase) {
+			// User cancelled
+			return;
+		}
+		databaseName = selectedDatabase.dbName;
+		if (selectedDatabase.isCreateNew) {
+			databaseName = await vscode.window.showInputBox(
+				{
+					title: constants.enterNewDatabaseName,
+					ignoreFocusOut: true
+				}
+			) ?? '';
+			// If user cancels out of this just return them to the db select quickpick in case they changed their mind
+		}
+	}
+
+
+	// 4. Modify sqlcmd vars
+	// TODO: Concat ones from publish profile
+	let sqlCmdVariables = Object.assign({}, project.sqlCmdVariables);
+
+	if (Object.keys(sqlCmdVariables).length > 0) {
+		// Loop over the vars to modify until the user is done
```

Comment:
Consider expanding a bit more in the comment on why this is an unbounded loop instead of just a for-loop.

> Continually loop over the sqlcmd variables until the user is done setting their values.  The user can modify values multiple times if they change their mind, or finish the operation midway if they don't want to manually override all sqlcmd variables.

or something like that.